disperser/batcher/batch_confirmer.go

```diff
@@ -0,0 +1,531 @@
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-stateUpdateTicker.C:
+			batch, minibatches, err = b.MinibatchStore.GetLatestFormedBatch(ctx)
+			if err != nil {
+				b.logger.Error("error getting latest formed batch", "err", err)
+			}
+		}
+	}
+
+	if len(minibatches) == 0 {
+		return fmt.Errorf("no minibatches in the batch %s", batch.ID)
+	}
+
+	// Make sure all minibatches in the batch have been dispersed
+	batchDispersed := false
+	stateUpdateTicker.Reset(b.DispersalStatusCheckInterval)
+	ctxWithTimeout, cancel := context.WithTimeout(ctx, b.DispersalTimeout)
+	defer cancel()
+	for !batchDispersed {
+		select {
+		case <-ctxWithTimeout.Done():
+			return ctxWithTimeout.Err()
+		case <-stateUpdateTicker.C:
+			batchDispersed, err = b.MinibatchStore.BatchDispersed(ctx, batch.ID)
+			if err != nil {
+				b.logger.Error("error checking if batch is dispersed", "err", err)
+			}
+		}
+	}
+
+	if !batchDispersed {
+		return errors.New("batch not dispersed")
+	}
+
+	// Get batch state
+	batchState := b.Minibatcher.PopBatchState(batch.ID)
+	if batchState == nil {
+		return fmt.Errorf("no batch state found for batch %s", batch.ID)
+	}
+
+	// Construct batch header
+	b.logger.Debug("Constructing batch header...")
+	batchHeader := &core.BatchHeader{
+		ReferenceBlockNumber: batch.ReferenceBlockNumber,
+		BatchRoot:            [32]byte{},
+	}
+	blobHeaderHashes := make([][32]byte, 0)
+	for _, minibatch := range minibatches {
+		blobHeaderHashes = append(blobHeaderHashes, minibatch.BlobHeaderHashes...)
+	}
+	merkleTree, err := batchHeader.SetBatchRootFromBlobHeaderHashes(blobHeaderHashes)
+	if err != nil {
+		return fmt.Errorf("error setting batch root from blob header hashes: %w", err)
+	}
+	batchHeaderHash, err := batchHeader.GetBatchHeaderHash()
+	if err != nil {
+		return fmt.Errorf("error getting batch header hash: %w", err)
+	}
+
+	// Make AttestBatch call
+	b.logger.Debug("Attesting batch...")
+	quorumIDsMap := make(map[core.QuorumID]struct{})
+	for _, blobHeader := range batchState.BlobHeaders {
+		for _, q := range blobHeader.QuorumInfos {
+			if _, ok := quorumIDsMap[q.QuorumID]; !ok {
+				quorumIDsMap[q.QuorumID] = struct{}{}
+			}
+		}
+	}
+	quorumIDs := make([]core.QuorumID, len(quorumIDsMap))
+	i := 0
+	for q := range quorumIDsMap {
+		quorumIDs[i] = q
+		i++
+	}
+
+	replyChan, err := b.Dispatcher.AttestBatch(ctx, batchState.OperatorState, blobHeaderHashes, batchHeader)
+	if err != nil {
+		return fmt.Errorf("error making attesting batch request: %w", err)
+	}
+
+	// Aggregate the signatures
+	b.logger.Debug("Aggregating signatures...")
+
+	quorumAttestation, err := b.Aggregator.ReceiveSignatures(ctx, batchState.OperatorState, batchHeaderHash, replyChan)
+	if err != nil {
+		_ = b.handleFailure(ctx, batchState.BlobMetadata, FailAggregateSignatures)
+		return fmt.Errorf("error receiving and validating signatures: %w", err)
+	}
+	operatorCount := make(map[core.QuorumID]int)
+	signerCount := make(map[core.QuorumID]int)
```

Comment:
Are the `signerCount` and `operatorCount` actually used?

Reply:
These are meant for reporting metrics, but there are no metrics on minibatching right now. 
I added a log line that prints these numbers for now.